src/main/java/mandel/rle/Rle.java

```diff
@@ -0,0 +1,98 @@
+package mandel.rle;
+
+public class Rle {
+    private int[][] grid;
+
+    public Rle(int rows, int cols) {
+        this.grid = new int[rows][cols];
+    }
+
+    public void loadFromRle(String rleContent) {
+        String[] lines = rleContent.split("\n");
+        StringBuilder rleInfo = new StringBuilder();
+
+        for (String line : lines) {
+            if (!line.startsWith("#") && !line.startsWith("x")) {
+                rleInfo.append(line);
+            }
+        }
+        decodeRle(rleInfo.toString());
+    }
+
+    public void decodeRle(String rle) {
+        int row = 0;
+        int col = 0;
+        int count = 0;
+
+        for (int i = 0; i < rle.length(); i++) {
+            char c = rle.charAt(i);
+            if (Character.isDigit(c)) {
+                count = Integer.parseInt(String.valueOf(c));;
+            } else {
+                if (count == 0) {
+                    count = 1;
+                }
+                switch (c) {
+                    case 'b':
+                        col += count;
+                        break;
+                    case 'o':
+                        for (int j = 0; j < count; j++) {
+                            grid[row][col] = 1;
+                            col++;
+                        }
+                        break;
+                    case '$':
+                        row += count;
+                        col = 0;
+                        break;
+                    case '!':
+                        return;
+                    default:
+                        System.out.println("Invalid character: " + c);
+                        break;
+                }
+               count = 0;
+            }
+        }
+    }
+
+    public int[][] getGrid() {
+        return grid;
+    }
+
+    public void nextGen()
```

Comment:
You already wrote this code once. Why did you copy it?

Reply:
I didn't want to delete all the code that I had written from the last assignment, so I added the nextGen() to this class and that's why I also added another frame and component. Should I delete the code from last week and overwrite the component and the frame so it only shows the glider when I run it?

Reply:
You don't need to delete any code. 

You can add the code to your existing code. 
You don't need a new frame or component.

Reply:
Oh ok that makes sense - I get it now, thank you